
# Ancestral State of Proteins  
  
## Overview  
  
The "Ancestral State of Proteins" project is dedicated to studying the evolutionary processes of protein metamorphism through phylogenetics. By analyzing the evolutionary lineage of proteins, this project aims to uncover how proteins have transformed and adapted through various evolutionary stages.  
  
## Objective  
  
The primary objective of this study is to understand the mechanisms behind protein evolution. By reconstructing the ancestral state of proteins, we can gain insights into the structural and functional adaptations that have occurred over millennia, helping to predict protein functions and interactions in a phylogenetic context.  
  
## Tools  
  
This project utilizes a suite of bioinformatics tools to perform sequence alignments, construct phylogenetic trees, and analyze evolutionary relationships:  
- **MAFFT**: A multiple sequence alignment program which is critical for aligning sequences in preparation for phylogenetic analysis.  
- **HMMER**: Utilized to handle and analyze sequence data using profile hidden Markov models, aiding in the identification and alignment of protein families.  
- **IQ-TREE**: A fast and effective stochastic algorithm to infer phylogenetic trees by maximum likelihood, providing insights into protein evolutionary pathways.  
  
## Installation  
  
To run the analyses, you will need to install various Python libraries and tools used in this project. All dependencies are listed in the `requirements.txt` file. To install these dependencies, please run the following command in your terminal:  
  
```bash 
pip3 install -r requirements.txt
```   

## Pipeline Steps

### Step 1: Data Preparation and Sequence Extraction

The first step involves preparing the data by loading configurations, input files, and executing functions to rename and process columns accordingly. This includes extracting amino acid sequences from PDB files based on identifiers provided in the input data. This step ensures the data is in the proper format for further analysis.

### Step 2: ...

...

## Usage  
  
Instructions on how to execute scripts and utilize the tools within this project will be further elaborated in subsequent sections of this README or accompanying documentation.
